cmd/block-generator/runner/run.go

```diff
@@ -82,33 +83,140 @@ func (r *Args) run() error {
+	record("_average", metrics.BlockUploadTimeName, rate)
+	record("_cumulative", metrics.BlockUploadTimeName, floatTotal)
+	record("", metrics.ImportedRoundGaugeName, intTotal)
+
+	if writeErrors.Len() > 0 {
+		return fmt.Errorf("error writing metrics (%s): %w", writeErrors.String(), writeErr)
+	}
+
+	// Calculate import transactions per second.
+	totalTxn, err := getMetric(entry, metrics.ImportedTxnsPerBlockName, false)
+	if err != nil {
+		return err
+	}
+
+	importTimeS, err := getMetric(entry, metrics.BlockImportTimeName, false)
+	if err != nil {
+		return err
+	}
+	tps := totalTxn / importTimeS
+	key := "overall_transactions_per_second"
+	msg := fmt.Sprintf("%s_%s:%.2f\n", prefix, key, tps)
+	if _, err := out.WriteString(msg); err != nil {
+		return fmt.Errorf("unable to write metric '%s': %w", key, err)
 	}
-	rate := sum / count
 
-	msg := fmt.Sprintf("%s:%f\n", key, rate)
+	// Uptime
+	key = "uptime_seconds"
+	msg = fmt.Sprintf("%s_%s:%.2f\n", prefix, key, time.Since(start).Seconds())
 	if _, err := out.WriteString(msg); err != nil {
 		return fmt.Errorf("unable to write metric '%s': %w", key, err)
 	}
+
 	return nil
 }
 
+func recordMetricToFile(entry Entry, outputKey, metricSuffix string, t metricType, out *os.File) error {
+	value, err := getMetric(entry, metricSuffix, t == rate)
+	if err != nil {
+		return err
+	}
+
+	var msg string
+	if t == intTotal {
+		msg = fmt.Sprintf("%s:%d\n", outputKey, uint64(value))
+	} else {
+		msg = fmt.Sprintf("%s:%.2f\n", outputKey, value)
+	}
+
+	if _, err := out.WriteString(msg); err != nil {
+		return fmt.Errorf("unable to write metric '%s': %w", outputKey, err)
+	}
+
+	return nil
+}
+
+func getMetric(entry Entry, suffix string, rateMetric bool) (float64, error) {
+	total := 0.0
+	sum := 0.0
+	count := 0.0
+	hasSum := false
+	hasCount := false
+	hasTotal := false
+
+	for _, metric := range entry.Data {
+		var err error
+
+		if strings.Contains(metric, suffix) {
+			split := strings.Split(metric, " ")
+			if len(split) != 2 {
+				return 0.0, fmt.Errorf("unknown metric format, expected 'key value' received: %s", metric)
+			}
+
+			// Check for _sum / _count for summary (rateMetric) metrics.
+			// Otherwise grab the total value.
+			if strings.HasSuffix(split[0], "_sum") {
+				sum, err = strconv.ParseFloat(split[1], 64)
+				hasSum = true
+			} else if strings.HasSuffix(split[0], "_count") {
+				count, err = strconv.ParseFloat(split[1], 64)
+				hasCount = true
+			} else if strings.HasSuffix(split[0], suffix) {
+				total, err = strconv.ParseFloat(split[1], 64)
+				hasTotal = true
+			}
+
+			if err != nil {
+				return 0.0, fmt.Errorf("unable to parse metric '%s': %w", metric, err)
+			}
+
+			if rateMetric && hasSum && hasCount {
+				return sum / count, nil
+			} else if !rateMetric {
+				if hasSum {
+					return sum, nil
+				}
+				if hasTotal {
+					return total, nil
+				}
+			}
+		}
+	}
+
+	return 0.0, fmt.Errorf("metric not found: %s", suffix)
```

Comment:
This may not be necessarily correct. 
Does it matter if the metric is there, but not in the form of rateMetric. And because the request is rateMetric, metric not found is returned. Does this mean "rate metric not found"?

Reply:
I think it's still correct, but you're right about there being some nuance to it... if the metric wasn't a "summary" then the rate fields wouldn't be there and the error could be more precise. On the other hand if the metric isn't there, it would also hit this  error.